pubtools/_quay/tag_docker.py

```diff
@@ -734,34 +722,46 @@ def run(self):
         self.check_input_validity()
         signature_handler = BasicSignatureHandler(self.hub, self.target_settings, self.target_name)
 
-        for item in self.push_items:
-            for tag in item.metadata["add_tags"]:
-                LOG.info("Processing add tag '{0}'".format(tag))
-                add_archs = self.tag_add_calculate_archs(item, tag)
-                # If all archs were somehow excluded from being added, no-op
-                if add_archs == []:
-                    LOG.warning("No archs can be added to tag '{0}', skipping".format(tag))
-                    continue
-                # If None, we're dealing with a source image and we want to copy to destination
-                elif add_archs is None:
-                    self.copy_tag_sign_images(item, tag, signature_handler)
-                # Otherwise, merge relevant archs of source and dest
-                else:
-                    self.merge_manifest_lists_sign_images(item, tag, add_archs, signature_handler)
-
-            for tag in item.metadata["remove_tags"]:
-                LOG.info("Processing remove tag '{0}'".format(tag))
-                remove_archs, keep_archs = self.tag_remove_calculate_archs(item, tag)
-                # If all archs were somehow excluded from removal, no-op
-                if not remove_archs:
-                    LOG.warning("No archs can be removed from tag '{0}', skipping".format(tag))
-                    continue
-                # If no archs will remain after removal, just perform untagging
-                elif not keep_archs:
-                    self.untag_image(item, tag)
-                # if some archs will be removed and some will remain, create new manifest list
-                else:
-                    self.manifest_list_remove_archs(item, tag, remove_archs)
+        with ContainerExecutor(
+            self.target_settings["skopeo_image"],
+            self.target_settings.get("docker_host") or "unix://var/run/docker.sock",
+            self.target_settings.get("docker_timeout"),
+            self.target_settings.get("docker_tls_verify") or False,
+            self.target_settings.get("docker_cert_path") or None,
+        ) as executor:
+            executor.skopeo_login(
+                self.target_settings["dest_quay_user"], self.target_settings["dest_quay_password"]
+            )
+            for item in self.push_items:
```

Comment:
Is there a chance for the `run()` method to include more conditions over time? In that case it may eventually become hard to understand even despite the pretty good documentation, so it would make sense to extract the loop body to separate method(s). Otherwise it's fine.

Reply:
I don't expect this logic to become more complex over time, but you never know. I'd say let's leave the refactoring out of this PR and do it once it becomes necessary. What do you think?